lib/viewmodels/comments_view_model.dart

```diff
@@ -45,6 +46,9 @@ class CommentsViewModel extends AutoDisposeFamilyAsyncNotifier<
           -commentA.publicationDate.compareTo(commentB.publicationDate),
     );
 
+    // Update the post comment count
+    ref.read(postCommentCountProvider(arg).notifier).setCount(comments.length);
```

Comment:
I am not sure why you call `setCount` here, and about the relevance of the `setCount` altogether.

Why do you need to set the precise count "without fetching the data" ?
I think Firebase already does caching and such optimizations, so we should simply call refresh to get the new data back.
Did you test this and found out it has measurable performance implications ?
I don't see any other viewmodel in the codebase where we applied such shortcuts of the database to avoid making a (probably local) Firebase query.

Reply:
You mentioned the `PostVotesViewModel` in the PR description but there it was not to be more efficient, it was from a user perspective to avoid directly showing a larger number change.

The comment count is not updated/visible directly when the user performs an action that could impact its value.
Also when the user posts a new comment we refetch all the comments, so the count will be updated with the actual value of the new number of comments in the database.

Reply:
My intuition was that the firestore cache could not be fully trusted because, when we go offline, a fetch to the database takes a time hinting that it tried to make an online fetch, which timed out, and then it looks in the cache. 
I however understand that this should be uniform in the code base, so I fixed it in 35238ccb5d910593b4f8d1c9ea6e6da1529eb188.

Reply:
exactly, ty :+1: